kubectl-fdb/cmd/restart.go

```diff
@@ -0,0 +1,203 @@
+/*
+ * restart.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+	"github.com/FoundationDB/fdb-kubernetes-operator/controllers"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func newRestartCmd(streams genericclioptions.IOStreams, rootCmd *cobra.Command) *cobra.Command {
+	o := NewFDBOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "restart",
+		Short: "Restarts process(es) in a given FDB cluster.",
+		Long:  "Restarts process(es) in a given FDB cluster.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			force, err := rootCmd.Flags().GetBool("force")
+			if err != nil {
+				return err
+			}
+			clusterName, err := cmd.Flags().GetString("fdb-cluster")
+			if err != nil {
+				return err
+			}
+			allProcesses, err := cmd.Flags().GetBool("all-processes")
+			if err != nil {
+				return err
+			}
+			allProcessesIncorrectCMD, err := cmd.Flags().GetBool("all-processes-with-incorrect-cmd")
```

Comment:
I wonder if there's a way we can make this a little more generic. Maybe we could allow filtering by any condition?

Reply:
I'm also worried we could find ourselves in a place where we have multiple different commands implementing their own process / pod filtering logic in a way that leads to duplication of effort.

Reply:
Making this more generic sounds good 👍 I already planned as a next step to review the current mechanism and to merge (at least `remove instances` and this one) with a similar logic.